test/sql/multi_transaction_index.sql

```diff
@@ -0,0 +1,144 @@
+
+-- Create index with long name to see how this is handled on chunks
+CREATE INDEX a_hypertable_index_with_a_very_very_long_name_that_truncates ON index_test (time, temp) WITH (hypertable.multi_transaction);
+CREATE INDEX a_hypertable_index_with_a_very_very_long_name_that_truncates_2 ON index_test (time, temp) WITH (hypertable.multi_transaction);
+
+SELECT * FROM test.show_indexes('index_test');
+SELECT * FROM test.show_indexesp('_timescaledb_internal._hyper%_chunk');
+
+DROP INDEX a_hypertable_index_with_a_very_very_long_name_that_truncates;
+DROP INDEX a_hypertable_index_with_a_very_very_long_name_that_truncates_2;
+
+
+SELECT * FROM test.show_indexes('index_test');
+SELECT * FROM test.show_indexesp('_timescaledb_internal._hyper%_chunk');
+
+SELECT * FROM test.show_indexes('index_test');
+SELECT * FROM test.show_indexesp('_timescaledb_internal._hyper%_chunk');
+
+-- Add constraint index
+ALTER TABLE index_test ADD UNIQUE (time, device);
+
+SELECT * FROM test.show_indexes('index_test');
+SELECT * FROM test.show_indexesp('_timescaledb_internal._hyper%_chunk');
+
+-- Constraint indexes are added to chunk_index table.
+SELECT * FROM _timescaledb_catalog.chunk_index;
+SELECT * FROM _timescaledb_catalog.chunk_constraint;
+
+DROP TABLE index_test;
+
+-- Metadata removed
+SELECT * FROM _timescaledb_catalog.chunk_index;
+
+-- Test that indexes are planned correctly
+CREATE TABLE index_expr_test(id serial, time timestamptz, temp float, meta int);
+select create_hypertable('index_expr_test', 'time');
+
+-- Screw up the attribute numbers
+ALTER TABLE index_expr_test DROP COLUMN id;
+
+CREATE INDEX ON index_expr_test (meta) WITH (hypertable.multi_transaction);
+INSERT INTO index_expr_test VALUES ('2017-01-20T09:00:01', 17.5, 1);
+INSERT INTO index_expr_test VALUES ('2017-01-20T09:00:01', 17.5, 2);
+
+
+SET enable_seqscan TO false;
+SET enable_bitmapscan TO false;
+EXPLAIN (verbose, costs off)
+SELECT * FROM index_expr_test WHERE meta = 1;
+SELECT * FROM index_expr_test WHERE meta = 1;
+SET enable_seqscan TO default;
+SET enable_bitmapscan TO default;
+
+\set ON_ERROR_STOP 0
+-- cannot create a multi_transaction index within a transaction block
+BEGIN;
+CREATE INDEX ON index_expr_test (temp) WITH (hypertable.multi_transaction);
+ROLLBACK;
+\set ON_ERROR_STOP 1
+
+DROP TABLE index_expr_test CASCADE;
+
+CREATE TABLE partial_index_test(time INTEGER);
+SELECT create_hypertable('partial_index_test', 'time', chunk_time_interval => 1, create_default_indexes => false);
+
+-- create 3 chunks
+INSERT INTO partial_index_test(time) SELECT generate_series(0, 2);
+
+select * from partial_index_test order by 1;
+
+-- create indexes on only 1 of the chunks
+CREATE INDEX ON partial_index_test (time) WITH (hypertable.multi_transaction, hypertable.max_chunks='1');
+
+SELECT * FROM test.show_indexes('partial_index_test');
+SELECT * FROM test.show_indexesp('_timescaledb_internal._hyper%_chunk');
+
+-- with broken ordered append, this caused an assertion failure when a
```

Comment:
i think you should remove `with broken ordered append` and maybe add issue/pr/commit that has details whats being tested here